code/mob.dm

```diff
@@ -2980,9 +2980,28 @@
 	if(!istype(A) || !user)
 		return FALSE // who's eating what, now?
 
+	/// Apparently you could feed things to ghosts. So spooky everyone loses their appetite
+	if (isobserver(src))
+		if(isAIeye(src))
+			src.can_not_eat(A, user, "is_aieye")
+		else
+			src.can_not_eat(A, user, "is_ghost")
+		return FALSE
+
+	/// Science still hasn't found a way to install a cyberstomach into a cyberperson
+	if (issilicon(src))
+		src.can_not_eat(A, user, "is_silicon")
+		return FALSE
+
+	/// And just in case we're something that isnt caught by the above and isnt something that's supposed to eat things
+	if (!isliving(src))
+		src.can_not_eat(A, user, null)
+		return FALSE
+
 	/// Making sure the thing actually exists and isn't cheating the bite-rate. And is also edible, so we don't get people trying to stuff multitools in their mouth
```

Comment:
```suggestion
	// Science still hasn't found a way to install a cyberstomach into a cyberperson
	if (issilicon(src))
		src.can_not_eat(A, user, "is_silicon")
		return FALSE

	// And just in case we're something that isnt caught by the above and isnt something that's supposed to eat things
	if (!isliving(src))
		src.can_not_eat(A, user, null)
		return FALSE

	// Making sure the thing actually exists and isn't cheating the bite-rate. And is also edible, so we don't get people trying to stuff multitools in their mouth
```